src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/Extensions/TypeExtensions.cs

```diff
@@ -45,6 +46,7 @@ public static class TypeExtensions
                 [typeof(double?)] = () => new OpenApiSchema {Type = "number", Format = "double", Nullable = true},
                 [typeof(decimal?)] = () => new OpenApiSchema {Type = "number", Format = "double", Nullable = true},
                 [typeof(DateTime?)] = () => new OpenApiSchema {Type = "string", Format = "date-time", Nullable = true},
+                [typeof(TimeSpan?)] = () => new OpenApiSchema {Type = "string", Nullable = true},
```

Comment:
>TimeSpan?) [](start = 24, length = 10)

add a unit test case or update existing to cover this datatype

Reply:
Will add this in, thanks! I couldn't locate existing unit tests for this class but it may be indirectly tested somewhere else; I'll do some digging.

Reply:
Most of the test call the generator method directly, you can update any existing xml used in test [https://github.com/microsoft/OpenAPI.NET.CSharpAnnotations/blob/master/test/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Tests/OpenApiDocumentGeneratorTests/OpenApiDocumentGeneratorTest.cs](https://github.com/microsoft/OpenAPI.NET.CSharpAnnotations/blob/master/test/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Tests/OpenApiDocumentGeneratorTests/OpenApiDocumentGeneratorTest.cs) and updated the expected json

---
In reply to: [377244341](https://github.com/microsoft/OpenAPI.NET.CSharpAnnotations/pull/230#discussion_r377244341) [](ancestors = 377244341)